clients/spark/core/src/main/scala/io/treeverse/gc/CommittedAddressLister.scala

```diff
@@ -1,35 +1,53 @@
 package io.treeverse.gc
 
-import org.apache.spark.sql.DataFrame
-import io.treeverse.clients.LakeFSContext
-import org.apache.spark.sql.SparkSession
-import io.treeverse.clients.LakeFSJobParams
+import io.treeverse.clients.{LakeFSContext, LakeFSJobParams}
 import io.treeverse.lakefs.catalog.Entry.AddressType
-import org.apache.spark.sql.functions.col
+import org.apache.spark.sql.{DataFrame, SparkSession}
 
 trait CommittedAddressLister {
-  def listCommittedAddresses(spark: SparkSession, storageNamespace: String): DataFrame
+  def listCommittedAddresses(
+      spark: SparkSession,
+      storageNamespace: String,
+      clientStorageNamespace: String
+  ): DataFrame
 }
 
 class NaiveCommittedAddressLister extends CommittedAddressLister {
-  override def listCommittedAddresses(spark: SparkSession, storageNamespace: String): DataFrame = {
+  override def listCommittedAddresses(
+      spark: SparkSession,
+      storageNamespace: String,
+      clientStorageNamespace: String
+  ): DataFrame = {
     val normalizedStorageNamespace =
       if (storageNamespace.endsWith("/")) storageNamespace else storageNamespace + "/"
     val params = LakeFSJobParams.forStorageNamespace(
       normalizedStorageNamespace,
       UncommittedGarbageCollector.UNCOMMITTED_GC_SOURCE_NAME
     )
-    var df = LakeFSContext.newDF(spark, params)
-    df = df
-      // TODO (optional): push down a filter to the input format, to filter out absolute addresses!
-      .filter(
-        (col("address_type") === AddressType.RELATIVE.name) ||
-          // Backwards compatability with entries prior to address_type
-          (col("address_type") === AddressType.BY_PREFIX_DEPRECATED.name &&
-            !col("address_type").contains("://"))
-      )
-      .select("address")
-      .distinct
+    val df = LakeFSContext.newDF(spark, params)
+
+    val normalizedClientStorageNamespace =
+      if (clientStorageNamespace.endsWith("/")) clientStorageNamespace
+      else clientStorageNamespace + "/"
+
+    import spark.implicits._
     df
+      .select("address_type", "address")
+      .filter(row => {
+        val (addrType, addr) = (row.getString(0), row.getString(1))
```

Comment:
The current code ignores address_type BY_PREFIX_DEPRECATED which are not in full path format.
BY_PREFIX_DEPRECATED can have 3 type of addresses:
1. Relative path in namespace
2. Full path in namespace
3. Full path outside of namespace
We need to take all under consideration

Reply:
Addressed the last path - type deprecated with relative path